packages/flutter/lib/src/material/text_field.dart

```diff
@@ -92,7 +92,11 @@ class _TextFieldSelectionGestureDetectorBuilder extends TextSelectionGestureDete
       switch (Theme.of(_state.context).platform) {
         case TargetPlatform.iOS:
         case TargetPlatform.macOS:
-          renderEditable.selectWordEdge(cause: SelectionChangedCause.tap);
+          if (details.kind == PointerDeviceKind.mouse) {
```

Comment:
anything that looks at enums should use `switch` and explicit cases, not `if` or `default`, per our style guide